packages/material-ui-lab/src/Breadcrumb/Breadcrumb.js

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { withStyles } from '@material-ui/core/styles';
+import { emphasize } from '@material-ui/core/styles/colorManipulator';
+import ButtonBase from '@material-ui/core/ButtonBase';
+
+const height = 24;
+
+const styles = theme => ({
+  button: {
+    fontFamily: theme.typography.fontFamily,
+    fontSize: theme.typography.fontSize,
+    height,
+    lineHeight: `${height}px`,
+    color: theme.palette.grey[800],
+    borderRadius: 2,
+    '&:hover, &:focus': {
+      backgroundColor: theme.palette.grey[100],
+    },
+    '&:active': {
+      boxShadow: theme.shadows[0],
+      backgroundColor: emphasize(theme.palette.grey[100], 0.12),
+    },
+  },
+  active: {
+    fontWeight: theme.typography.fontWeightMedium,
+  },
+  gutters: {
+    paddingLeft: 8,
+    paddingRight: 8,
+  },
+  icon: {
+    marginRight: 4,
+    width: height / 1.2,
+    height: height / 1.2,
+    color:
+      theme.palette.type === 'light'
+        ? theme.palette.grey[700]
+        : theme.palette.grey[300],
+  },
+});
+
+function DefaultComponent(props) {
+  const { className, label, icon, active, tabIndex, onClick, ...rest } = props;
+
+  return (
+    <ButtonBase
+      variant="default"
+      color="primary"
+      className={className}
+      tabIndex={tabIndex}
+      onClick={onClick}
+      disabled={active}
+      {...rest}
+    >
+      {icon}
+      {label}
+    </ButtonBase>
+  );
+}
+
+class Breadcrumb extends React.PureComponent {
+  render() {
+    const {
+      classes,
+      className,
+      component: Component,
+      icon: iconProp,
+      label,
+      active,
+      disableGutters,
+      tabIndex,
+      onClick,
+      ...other
+    } = this.props;
+
+    let newTabIndex = tabIndex;
+
+    if (!newTabIndex) {
+      newTabIndex = onClick ? 0 : -1;
+    }
+
+    let icon = null;
+    if (iconProp && React.isValidElement(iconProp)) {
```

Comment:
nit: Checking if `iconProp` is falsy is redundant. `React.isValidElement` is already doing this: https://github.com/facebook/react/blob/v16.3.0/packages/react-is/src/__tests__/ReactIs-test.js#L62-L67

`react-is` uses `React.isValidElementType`. Or rather both share the same code.